Civi/Api4/Service/Spec/Provider/ActionScheduleSpecProvider.php

```diff
@@ -29,8 +29,8 @@ public function modifySpec(RequestSpec $spec) {
       $spec->getFieldByName('name')->setRequired(FALSE);
       // Repeat events do not require mapping_id or start_action_unit (or seemingly start_action_condition)- although
       // we don't have that level of nuance available so we make them optional for all events.
-      $spec->getFieldByName('mapping_id')->setRequiredIf('empty($values.used_for) || $values.used_for !== "civicrm_event"');
-      $spec->getFieldByName('start_action_unit')->setRequiredIf('empty($values.absolute_date) && (empty($values.used_for) || $values.used_for !== "civicrm_event")');
+      $spec->getFieldByName('mapping_id')->setRequiredIf('empty($values.used_for)');
+      $spec->getFieldByName('start_action_unit')->setRequiredIf('empty($values.absolute_date) && empty($values.start_action_date)');
```

Comment:
I guess a case could be made to loosen only in the rc & tighten in master - specifically regarding `start_action_date` - what do you think?

Reply:
@eileenmcnaughton you mean replace the && with || for start_action_unit here ?

Reply:
@monishdeb I guess I'm wondering if we should not set `start_action_unit` to required at all in stable / the rc - and do it in master - but if you feel confident I'm OK with merging as is

Reply:
Hmm on second thought when I looked on DB entries in civicrm_action_schedule table, for repeat activity and event, the start_action_unit is NULL so this requiredif on start_action_unit, doesn't have any affect. So I'll drop the requiredif.